libkbfs/md_ops.go

```diff
@@ -370,61 +371,43 @@ func (md *MDOpsStandard) processRange(ctx context.Context, id tlf.ID,
 		return nil, nil
 	}
 
-	var wg sync.WaitGroup
-	numWorkers := len(rmdses)
-	if numWorkers > maxMDsAtATime {
-		numWorkers = maxMDsAtATime
-	}
-	wg.Add(numWorkers)
+	var eg errgroup.Group
 
 	// Parallelize the MD decryption, because it could involve
 	// fetching blocks to get unembedded block changes.
 	rmdsChan := make(chan *RootMetadataSigned, len(rmdses))
 	irmdChan := make(chan ImmutableRootMetadata, len(rmdses))
-	errChan := make(chan error, 1)
 	var getRangeLock sync.Mutex
-	worker := func() {
-		defer wg.Done()
+	worker := func() error {
 		for rmds := range rmdsChan {
 			extra, err := md.getExtraMD(ctx, rmds.MD)
 			if err != nil {
-				select {
-				case errChan <- err:
-				default:
-				}
-				return
+				return err
```

Comment:
Actually, no need -- all the calls do RPCs, and those already check the context.